src/mem/epoch/markable_atomic.rs

```diff
@@ -5,33 +5,38 @@ use std::sync::atomic::{self, Ordering};
 
 use super::{Owned, Shared, Guard};
 
-/// Like `std::sync::atomic::AtomicPtr`.
+/// Provides atomic access to a `(*const T, usize)` pair.
 ///
-/// Provides atomic access to a (nullable) pointer of type `T`, interfacing with
-/// the `Owned` and `Shared` types.
+/// The pair consists of:
+///
+/// - a (nullable) pointer of type `T`, interfacing with the `Owned` and `Shared` types
+///
+/// - a `usize` marker value that can fit into the unused bits of the pointer:
+/// `mark < std::mem::align_of::<T>()`
 #[derive(Debug)]
-pub struct MarkedAtomic<T> {
+pub struct MarkableAtomic<T> {
```

Comment:
I see your point, but I think `Markable` only makes sense for a boolean tag, where the states are "marked" and "unmarked." In the more general case, I think it makes more sense to think of it as a pointer that always has a mark, just with several possible mark values.

That said, you are right that `Marked` could be confusing. Maybe `Tagged`?